gym/utils/env_checker.py

```diff
@@ -126,81 +126,92 @@ def check_reset_seed(env: gym.Env):
-                obs in env.observation_space
-            ), "The value returned by `env.reset(return_info=True)` is not within the observation space."
-
-            result = env.reset(return_info=True)
-            assert isinstance(
-                result, tuple
-            ), f"Calling the reset method with `return_info=True` did not return a tuple, actual type: {type(result)}"
-            assert (
-                len(result) == 2
-            ), f"Calling the reset method with `return_info=True` did not return a 2-tuple, actual length: {len(result)}"
-
-            obs, info = result
-            assert (
-                obs in env.observation_space
-            ), "The first element returned by `env.reset(return_info=True)` is not within the observation space."
-            assert isinstance(
-                info, dict
-            ), f"The second element returned by `env.reset(return_info=True)` was not a dictionary, actual type: {type(info)}"
+            env.reset(options={})
         except TypeError as e:
             raise AssertionError(
-                "The environment cannot be reset with `return_info=True`, even though `return_info` or `kwargs` appear in the signature. "
+                "The environment cannot be reset with options, even though `options` or `**kwargs` appear in the signature. "
                 f"This should never happen, please report this issue. The error was: {e}"
             )
     else:
         raise gym.error.Error(
-            "The `reset` method does not provide a `return_info` or `**kwargs` keyword argument."
+            "The `reset` method does not provide an `options` or `**kwargs` keyword argument."
         )
 
 
-def check_reset_options(env: gym.Env):
-    """Check that the environment can be reset with options.
+def check_reset_return_info_deprecation(env: gym.Env):
+    """Makes sure support for deprecated `return_info` argument is dropped.
 
     Args:
         env: The environment to check
-
     Raises:
-        AssertionError: The environment cannot be reset with options,
-            even though `options` or `kwargs` appear in the signature.
+        AssertionError depending on spec violation
     """
     signature = inspect.signature(env.reset)
-    if "options" in signature.parameters or (
-        "kwargs" in signature.parameters
-        and signature.parameters["kwargs"].kind is inspect.Parameter.VAR_KEYWORD
-    ):
-        try:
-            env.reset(options={})
-        except TypeError as e:
-            raise AssertionError(
-                "The environment cannot be reset with options, even though `options` or `**kwargs` appear in the signature. "
-                f"This should never happen, please report this issue. The error was: {e}"
-            )
-    else:
-        raise gym.error.Error(
-            "The `reset` method does not provide an `options` or `**kwargs` keyword argument."
+    if "return_info" in signature.parameters:
+        raise AssertionError(
```

Comment:
I think it's better to make this a warning - it is an important information, but at the same time, in principle, it's not incorrect to implement a custom environment which takes extra optional arguments.

Reply:
Sounds good, I'm changing this to a UserWarning.